packages/compass-aggregations/src/components/aggregation-side-panel/stage-wizard-use-cases/lookup/lookup.tsx

```diff
@@ -0,0 +1,191 @@
+import {
+  Body,
+  spacing,
+  css,
+  TextInput,
+  Combobox,
+  ComboboxWithCustomOption,
+  ComboboxOption,
+} from '@mongodb-js/compass-components';
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
+import { connect } from 'react-redux';
+import type { RootState } from '../../../../modules';
+import { fetchCollectionFields } from '../../../../modules/collections-fields';
+import type { CollectionData } from '../../../../modules/collections-fields';
+
+type LookupFormState = {
+  from: string;
+  localField: string;
+  foreignField: string;
+  as: string;
+};
+
+const containerStyles = css({
+  display: 'flex',
+  flexDirection: 'column',
+  gap: spacing[2],
+});
+
+const titleStyles = css({
+  minWidth: `${'Join documents in'.length}ch`,
+  textAlign: 'right',
+});
+
+const formGroup = css({
+  display: 'flex',
+  gap: spacing[2],
+  alignItems: 'center',
+});
+
+const inputFieldStyles = css({
+  width: '300px',
+});
+
+export const LookupForm = ({
+  fields,
+  collectionsFields,
+  onSelectCollection,
+  onChange,
+}: {
+  fields: string[];
+  collectionsFields: Record<string, CollectionData>;
+  onSelectCollection: (collection: string) => void;
+  onChange: (value: string, error: Error | null) => void;
+}) => {
+  const [formData, setFormData] = useState<LookupFormState>({
+    as: '',
+    from: '',
+    foreignField: '',
+    localField: '',
+  });
+
+  useEffect(() => {
+    if (formData.from) {
+      onSelectCollection(formData.from);
+    }
+  }, [formData.from, onSelectCollection]);
```

Comment:
This `useEffect` and the one below it look to me to be a bit circuitous. Is there a reason we don't call the `onSelectCollection` directly when it's changed in `onSelectOption`?

Reply:
nice catch. added in d71dfbc0e2de21067a67d1526925bbd6896dafa3